ZipArchive.cs

```diff
@@ -248,20 +376,115 @@ public ZipEntry AddFile (string sourcePath, string archivePath = null, EntryPerm
 			return ReadEntry ((ulong)index);
 		}
 
+		/// <summary>
+		/// Adds the stream to the archive.
+		/// </summary>
+		/// <param name="entryName">The name of the entry with in the archive</param>
+		/// <param name="data">A stream containing the data to add to the archive</param>
+		/// <param name="method">The compression method to use</param>
+		public void AddEntry (string entryName, Stream data,
+			CompressionMethod method = CompressionMethod.DEFAULT)
+		{
+			AddStream (data, entryName, method: method);
+		}
+
+		/// <summary>
+		/// Adds the byte[] data to the archive
+		/// </summary>
+		/// <param name="entryName">The name of the entry with in the archive</param>
+		/// <param name="data">The draw byte[] array to add to the entry</param>
+		/// <param name="method">The compression method to use</param>
+		public void AddEntry (string entryName, byte [] data,
```

Comment:
I don't think we need this overload. The existing `AddEntry` for `byte[]` is good enough.